tracee-ebpf/tracee/tracee.go

```diff
@@ -1493,6 +1673,46 @@ func (t *Tracee) processFileWrites() {
 	}
 }
 
+func (t *Tracee) processNetEvents() {
+	// Todo: split pcap files by context (tid + comm)
+	// Todo: add stats for network packets (in epilog)
+	// Todo: support syn+syn-ack packets
+	for {
+		select {
+		case in := <-t.netChannel:
+			// Sanity check - timestamp, host tid, comm and packet length must exist in all net events
+			if len(in) < 32 {
+				continue
+			}
+
+			timeStamp := binary.LittleEndian.Uint64(in[0:8])
+			pktLen := binary.LittleEndian.Uint32(in[8:12])
+
+			info := gopacket.CaptureInfo{
+				Timestamp:      time.Unix(0, int64(timeStamp)),
```

Comment:
I'm getting:

```
$ tcpdump -n -r ./capture.pcap
reading from file ./capture.pcap, link-type EN10MB (Ethernet)
-2:-41:-32.988136 IP 127.0.0.1.54207 > 127.0.0.1.8080: UDP, length 5
-2:-41:-32.988136 IP 127.0.0.1.54207 > 127.0.0.1.8080: UDP, length 5
-2:-41:-32.988137 IP 127.0.0.1.8080 > 127.0.0.1.54207: UDP, length 5
-2:-41:-32.988137 IP 127.0.0.1.8080 > 127.0.0.1.54207: UDP, length 5
-2:-41:-31.055386 IP 127.0.0.1.33117 > 127.0.0.1.8080: UDP, length 5
-2:-41:-31.055386 IP 127.0.0.1.33117 > 127.0.0.1.8080: UDP, length 5
-2:-41:-31.056352 IP 127.0.0.1.8080 > 127.0.0.1.33117: UDP, length 5
-2:-41:-31.056352 IP 127.0.0.1.8080 > 127.0.0.1.33117: UDP, length 5
```

this timeStamp seems broken. Trying to check what could be wrong I saw:

pkt.ts = bpf_ktime_get_ns()/1000;

You are transforming the nanosec into microsec in tracee.bpf.c -> tc_probe. 

I think It should be kept as nanosec as:

```
timeStamp := time.Unix(0, timeStamp))
```

will have ns as the second argument as nanosec:

```
func Unix(sec int64, nsec int64) Time {
```

I have believe it is right to consider it is little endian as this is not coming from the network stack but from task_struct.

Reply:
Working fine now:

```
14:14:34.409831 IP 127.0.0.1.53740 > 127.0.0.1.12345: Flags [.], ack 1741000412, win 512, options [nop,nop,TS val 1494266592 ecr 1494266592], length 0
14:14:34.409841 IP 127.0.0.1.53740 > 127.0.0.1.12345: Flags [.], ack 1, win 512, options [nop,nop,TS val 1494266592 ecr 1494266592], length 0
14:14:34.880155 IP 127.0.0.1.53740 > 127.0.0.1.12345: Flags [P.], seq 0:5, ack 1, win 512, options [nop,nop,TS val 1494267063 ecr 1494266592], length 5
```